From 2aaae14f9a20a4ee9b1fefee67c79ea2f943dfb3 Mon Sep 17 00:00:00 2001
From: "Ross-Ashikyan, William" <washikyan@maxlinear.com>
Date: Mon, 30 Sep 2024 10:31:17 -0700
Subject: [PATCH] WLANRTSYS-84783 Extend TPE usage to Indoor Standard Power

Issue: TPE IE was not seen for Indoor Standard Power

RC: TPE was orginally configured to only LPI and SP

Fix: Change check for SP to generic one for both SP and Indoor SP

FIXUP to WLANRTSYS-82700 mxl changes for V7[4/4] AFC  patches
---
 src/ap/mxl_ieee802_11.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ap/mxl_ieee802_11.c b/src/ap/mxl_ieee802_11.c
index 01beb58c1..57c774ef1 100644
--- a/src/ap/mxl_ieee802_11.c
+++ b/src/ap/mxl_ieee802_11.c
@@ -120,7 +120,7 @@ u8 * hostapd_eid_txpower_envelope_vendor_mxl_6g(struct hostapd_data *hapd, u8 *e
 	}
 
 
-	if (hapd->iface->conf->he_6ghz_reg_pwr_type == HE_REG_INFO_6GHZ_AP_TYPE_SP) {
+	if (he_reg_is_sp(iface->conf->he_6ghz_reg_pwr_type)) {
 		width = op_class_to_bandwidth(hapd->iface->conf->op_class);
 		switch (width) {
 			case 20:
-- 
2.43.0

